test/Mono.Linker.Tests.Cases/Reflection/TypeUsedViaReflection.cs

```diff
@@ -25,6 +29,14 @@ public static void Main ()
 			TestDeeplyNested ();
 			TestTypeOf ();
 			TestTypeFromBranch (3);
+			TestTypeUsingCaseInsensitiveFlag ();
+			TestTypeUsingCaseUnknownByTheLinker ();
+			/*
+			There is a test hole that involves overload Type.GetType(String, Func<AssemblyName,Assembly>, Func<Assembly,String,Boolean,Type>)
+			This couldn't be tested since Func<> create functions that cannot be marked with [Kept] and the test will end up generating an 
+			error message stating: Type `Mono.Linker.Tests.Cases.Reflection.TypeUsedViaReflection/<>c' should have been removed
+			The test case was validated manually and is commented in this file as function TestTypeOverloadWith3Parameters
```

Comment:
Will it work if instead of lambdas you use either static or local functions? Those can be annotated. Basically you want to avoid creating closures (lambdas do that), since it generates classes and code in the compiler. But you can create a Func over a normal method or local function (local functions also have closures, but it should be possible to add attributes to them).